src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -237,6 +239,69 @@ public Locatable getLocation() {
         return counts;
     }
 
+    /**
+     * Fixes the quality of all the elements that come from an overlapping pair in the same way as
+     * samtools does {@see tweak_overlap_quality function in
+     * <a href="https://github.com/samtools/htslib/blob/master/sam.c">samtools</a>}.
+     *
+     * Setting the quality of one of the bases to 0 effectively removes the redundant base for
+     * calling. In addition, if the bases overlap we have increased confidence if they agree (or
+     * reduced if they don't). Thus, the algorithm proceeds as following:
+     *
+     * 1. If the bases are the same, the quality of the first element is the sum of both qualities
+     *    and the quality of the second is reduced to 0.
+     * 2. If the bases are different, the base with the highest quality is reduced with a factor of
```

Comment:
There is no justification, just the samtools behaviour that I'm trying to mimic here.

My work is related in avoid the step of store the mpileup from samtools, that's why I need this as it is. Actually there are other method in `FragmentUtils` to correct for overlapping read pairs, and probably this will be more suitable for some API users. I can add in the javadoc a link to that method to note that there are other strategies to perform this.